--- a/meson.build
+++ b/meson.build
@@ -203,7 +203,10 @@
 
 # Optional dependency on GObject-Introspection; if GObject is disabled
 # then we don't build introspection data either
-build_gir = build_gobject and get_option('introspection') and get_option('default_library') == 'shared'
+build_gir = false
+if get_option('introspection') and get_option('default_library') == 'shared'
+  build_gir = true
+endif
 
 # Check for InitOnce on Windows
 if host_system == 'windows'
