--- a/meson.build	2019-08-13 20:56:43.000000000 -0500
+++ b/meson.build	2019-08-14 10:14:40.241944005 -0500
@@ -119,12 +119,13 @@
 # Check for session tracking backend
 session_tracking = 'null backend'
 
+enable_elogind = get_option('elogind')
 enable_systemd = get_option('systemd')
 enable_systemd_session = get_option('systemd_session') != 'disable'
 use_systemd_session = get_option('systemd_session') == 'default'
 enable_systemd_journal = get_option('systemd_journal')
 enable_consolekit = get_option('consolekit')
-if enable_systemd or enable_consolekit
+if enable_systemd or enable_consolekit or enable_elogind
   session_bin_deps += dependency('gio-unix-2.0', version: glib_req_version)
 
   # Check for systemd
@@ -149,6 +150,13 @@
     session_tracking = 'systemd'
   endif
 
+  # Check for elogind
+  if enable_elogind
+    libelogind_dep = dependency('libelogind', version: '>= 241', required: false)
+    session_bin_deps += libelogind_dep
+    session_tracking = 'elogind'
+  endif
+
   # Check for ConsoleKit
   if enable_consolekit
     dbus_glib_dep = dependency('dbus-glib-1', version: '>= 0.76')
@@ -156,7 +164,7 @@
 
     session_bin_deps += dbus_glib_dep
 
-    if enable_systemd
+    if enable_systemd or enable_elogind
       session_tracking += ' (with fallback to ConsoleKit)'
     else
       session_tracking = 'ConsoleKit'
@@ -166,6 +174,7 @@
 if enable_systemd_session
   assert(enable_systemd, 'Systemd support must be enabled when using systemd session management')
 endif
+config_h.set('HAVE_ELOGIND', enable_elogind)
 config_h.set('HAVE_SYSTEMD', enable_systemd)
 config_h.set('ENABLE_SYSTEMD_SESSION', enable_systemd_session)
 config_h.set('ENABLE_SYSTEMD_JOURNAL', enable_systemd_journal)
--- a/meson_options.txt	2019-08-13 20:56:43.000000000 -0500
+++ b/meson_options.txt	2019-08-14 10:17:45.902143845 -0500
@@ -1,8 +1,9 @@
 option('deprecation_flags', type: 'boolean', value: false, description: 'use *_DISABLE_DEPRECATED flags')
 option('session_selector', type: 'boolean', value: false, description: 'enable building a custom session selector dialog')
-option('systemd', type: 'boolean', value: true, description: 'Use systemd')
-option('systemd_session', type: 'combo', choices: ['disable', 'enable', 'default'], value: 'default', description: 'Whether to include systemd session support and use it by default')
-option('systemd_journal', type: 'boolean', value:'true', description: 'Send log to systemd-journal')
+option('elogind', type: 'boolean', value: true, description: 'Use elogind')
+option('systemd', type: 'boolean', value: false, description: 'Use systemd')
+option('systemd_session', type: 'combo', choices: ['disable', 'enable', 'default'], value: 'disable', description: 'Whether to include systemd session support and use it by default')
+option('systemd_journal', type: 'boolean', value:'false', description: 'Send log to systemd-journal')
 option('consolekit', type: 'boolean', value: false, description: 'Use consolekit')
 option('docbook', type: 'boolean', value: true, description: 'build documentation')
 option('man', type: 'boolean', value: true, description: 'build documentation (requires xmlto)')
--- a/gnome-session/gsm-systemd.c	2019-08-13 20:56:43.000000000 -0500
+++ b/gnome-session/gsm-systemd.c	2019-08-14 10:19:55.694541276 -0500
@@ -21,7 +21,7 @@
 #include "config.h"
 #include "gsm-systemd.h"
 
-#ifdef HAVE_SYSTEMD
+#if defined(HAVE_SYSTEMD) || defined(HAVE_ELOGIND)
 
 #include <errno.h>
 #include <string.h>
