--- a/configure.ac	2018-02-22 14:37:09.000000000 -0600
+++ b/configure.ac	2018-04-17 14:10:59.360041323 -0500
@@ -49,17 +49,17 @@
 AC_DISABLE_STATIC
 LT_INIT([win32-dll])
 
-AC_SUBST([PANGOMM_MODULES], ['glibmm-2.56 >= 2.55.1 cairomm-1.16 >= 1.2.2 pangocairo >= 1.38.0'])
+AC_SUBST([PANGOMM_MODULES], ['glibmm-2.58 >= 2.55.1 cairomm-1.16 >= 1.2.2 pangocairo >= 1.38.0'])
 PKG_CHECK_MODULES([PANGOMM], [$PANGOMM_MODULES])
 
 MM_PKG_CONFIG_SUBST([GTHREAD_CFLAGS], [--cflags-only-other gthread-2.0])
-MM_PKG_CONFIG_SUBST([GMMPROC_DIR], [--variable=gmmprocdir glibmm-2.56])
+MM_PKG_CONFIG_SUBST([GMMPROC_DIR], [--variable=gmmprocdir glibmm-2.58])
 
 MM_ARG_DISABLE_DEPRECATED_API
 MM_ARG_ENABLE_DOCUMENTATION
 MM_ARG_WITH_TAGFILE_DOC([libstdc++.tag], [mm-common-libstdc++])
 MM_ARG_WITH_TAGFILE_DOC([libsigc++-3.0.tag], [sigc++-3.0])
-MM_ARG_WITH_TAGFILE_DOC([glibmm-2.56.tag], [glibmm-2.56])
+MM_ARG_WITH_TAGFILE_DOC([glibmm-2.58.tag], [glibmm-2.58])
 MM_ARG_WITH_TAGFILE_DOC([cairomm-1.16.tag], [cairomm-1.16])
 
 AC_LANG([C++])


--- a/configure	2018-04-17 14:16:27.098643191 -0500
+++ b/configure	2018-04-17 14:19:10.223570778 -0500
@@ -17237,7 +17237,7 @@
 
 
 
-PANGOMM_MODULES='glibmm-2.56 >= 2.55.1 cairomm-1.16 >= 1.2.2 pangocairo >= 1.38.0'
+PANGOMM_MODULES='glibmm-2.58 >= 2.55.1 cairomm-1.16 >= 1.2.2 pangocairo >= 1.38.0'
 
 
 
@@ -17465,7 +17465,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GMMPROC_DIR" >&5
 $as_echo_n "checking for GMMPROC_DIR... " >&6; }
 if test -z "${GMMPROC_DIR+set}"; then :
-  GMMPROC_DIR=`$PKG_CONFIG --variable=gmmprocdir glibmm-2.56 2>&5`
+  GMMPROC_DIR=`$PKG_CONFIG --variable=gmmprocdir glibmm-2.58 2>&5`
        if test "$?" -eq 0; then :
 
 fi
@@ -17902,12 +17902,12 @@
     mm_htmlrefdir=`expr "X@$withval" : '.*@\(.*\)' 2>&5`
     mm_tagname=`expr "X/$withval" : '[^@]*[\\/]\([^\\/@]*\)@' 2>&5`
     mm_tagpath=`expr "X$withval" : 'X\([^@]*\)@' 2>&5`
-    test "x$mm_tagname" != x || mm_tagname="glibmm-2.56.tag"
+    test "x$mm_tagname" != x || mm_tagname="glibmm-2.58.tag"
     test "x$mm_tagpath" != x || mm_tagpath=$mm_tagname
 else
 
     mm_htmlrefdir=
-    mm_tagname="glibmm-2.56.tag"
+    mm_tagname="glibmm-2.58.tag"
     mm_tagpath=$mm_tagname
 fi
 
@@ -17922,7 +17922,7 @@
   # If no local directory was specified, get the default from the .pc file
   if test "x$mm_htmlrefdir" = x; then :
 
-    mm_htmlrefdir=`$PKG_CONFIG --variable=htmlrefdir "glibmm-2.56" 2>&5`
+    mm_htmlrefdir=`$PKG_CONFIG --variable=htmlrefdir "glibmm-2.58" 2>&5`
 fi
   # If the user specified a Web URL, allow it to override the public location
   case $mm_htmlrefdir in #(
@@ -17930,7 +17930,7 @@
     mm_htmlrefpub=$mm_htmlrefdir ;; #(
   *) :
 
-    mm_htmlrefpub=`$PKG_CONFIG --variable=htmlrefpub "glibmm-2.56" 2>&5`
+    mm_htmlrefpub=`$PKG_CONFIG --variable=htmlrefpub "glibmm-2.58" 2>&5`
     test "x$mm_htmlrefpub" != x || mm_htmlrefpub=$mm_htmlrefdir
     test "x$mm_htmlrefdir" != x || mm_htmlrefdir=$mm_htmlrefpub
    ;;
@@ -17941,7 +17941,7 @@
      ;; #(
   *) :
 
-    mm_doxytagfile=`$PKG_CONFIG --variable=doxytagfile "glibmm-2.56" 2>&5`
+    mm_doxytagfile=`$PKG_CONFIG --variable=doxytagfile "glibmm-2.58" 2>&5`
     test "x$mm_doxytagfile" = x || mm_tagpath=$mm_doxytagfile
    ;;
 esac
@@ -17953,8 +17953,8 @@
 $as_echo "$mm_tagpath@$mm_htmlrefdir" >&6; }
 
   if test "x$USE_MAINTAINER_MODE" != xno && test ! -f "$mm_tagpath"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Doxygen tag file glibmm-2.56.tag not found" >&5
-$as_echo "$as_me: WARNING: Doxygen tag file glibmm-2.56.tag not found" >&2;}
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Doxygen tag file glibmm-2.58.tag not found" >&5
+$as_echo "$as_me: WARNING: Doxygen tag file glibmm-2.58.tag not found" >&2;}
 fi
   if test "x$mm_htmlrefdir" = x; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Location of external glibmm documentation not set" >&5


--- a/tools/Makefile.am	2018-01-13 14:37:45.000000000 -0600
+++ b/tools/Makefile.am	2018-04-17 14:11:17.708142789 -0500
@@ -30,7 +30,7 @@
 noinst_PROGRAMS  = extra_defs_gen/generate_extra_defs
 
 extra_defs_gen_generate_extra_defs_SOURCES = extra_defs_gen/generate_defs_pango.cc
-extra_defs_gen_generate_extra_defs_LDADD   = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.56
+extra_defs_gen_generate_extra_defs_LDADD   = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.58
 extra_defs_gen_generate_extra_defs_LDFLAGS = -no-undefined -avoid-version
 
 AM_CPPFLAGS = -I$(top_builddir) $(GTHREAD_CFLAGS) $(PANGOMM_CFLAGS)


--- a/tools/Makefile.in	2018-02-22 14:37:57.000000000 -0600
+++ b/tools/Makefile.in	2018-04-17 14:30:13.555226530 -0500
@@ -374,7 +374,7 @@
 dist_gmmproc_m4_DATA = $(files_codegen_m4:%=m4/%)
 dist_noinst_DATA = README
 extra_defs_gen_generate_extra_defs_SOURCES = extra_defs_gen/generate_defs_pango.cc
-extra_defs_gen_generate_extra_defs_LDADD = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.56
+extra_defs_gen_generate_extra_defs_LDADD = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.58
 extra_defs_gen_generate_extra_defs_LDFLAGS = -no-undefined -avoid-version
 AM_CPPFLAGS = -I$(top_builddir) $(GTHREAD_CFLAGS) $(PANGOMM_CFLAGS)
 AM_CXXFLAGS = $(PANGOMM_WXXFLAGS)
