--- a/data/pam-exherbo/gdm-launch-environment.pam	2019-02-21 13:44:22.000000000 -0600
+++ b/data/pam-exherbo/gdm-launch-environment.pam	2019-02-28 15:25:16.456946612 -0600
@@ -8,7 +8,7 @@
 
 password    required    pam_deny.so
 
--session    optional    pam_systemd.so
+-session    optional    pam_elogind.so
 session     optional    pam_keyinit.so force revoke
 session     required    pam_succeed_if.so audit quiet_success user = gdm
 session     required    pam_permit.so
