--- a/configure	2018-05-07 12:59:35.962114638 -0500
+++ b/configure	2018-05-07 13:43:30.237252279 -0500
@@ -20233,7 +20233,7 @@
 
 # Check whether --with-ucd-dir was given.
 if test "${with_ucd_dir+set}" = set; then :
-  withval=$with_ucd_dir; UCD_DIR=$with_emoji_annotation_dir
+  withval=$with_ucd_dir; UCD_DIR=$with_ucd_dir
 else
   UCD_DIR="/usr/share/unicode/ucd"


--- a/configure.ac	2018-03-02 02:24:26.000000000 -0600
+++ b/configure.ac	2018-05-07 13:50:09.542380813 -0500
@@ -666,7 +666,7 @@
     AS_HELP_STRING([--with-ucd-dir[=DIR]],
         [Set the directory of UCD (Unicode Character Database) files.
          (default: "/usr/share/unicode/ucd")]),
-    UCD_DIR=$with_emoji_annotation_dir,
+    UCD_DIR=$with_ucd_dir,
     UCD_DIR="/usr/share/unicode/ucd"
 )
 AC_SUBST(UCD_DIR)
