diff -urN ibus-1.5.20.20190629/bindings/pygobject/Makefile.am ibus-1.5.20.20190629-fixed/bindings/pygobject/Makefile.am
--- ibus-1.5.20.20190629/bindings/pygobject/Makefile.am	2019-07-18 17:32:23.713358221 -0600
+++ ibus-1.5.20.20190629-fixed/bindings/pygobject/Makefile.am	2019-07-18 17:34:06.175848521 -0600
@@ -29,6 +29,7 @@
 overrides_PYTHON =				\
 	gi/overrides/IBus.py			\
 	$(NULL)
+endif
 
 TESTS =
 
