--- a/meson.build	2018-08-12 22:47:39.477938160 -0500
+++ b/meson.build	2018-08-12 22:47:51.149931809 -0500
@@ -1800,9 +1800,9 @@
   glib_conf.set('HAVE_STRLCPY', 1)
 endif

-python = import('python').find_installation('python3')
+python = import('python').find_installation('python')
 # used for '#!/usr/bin/env <name>'
-python_name = 'python3'
+python_name = 'python'

 # Determine which user environment-dependent files that we want to install
 have_bash = find_program('bash', required : false).found() # For completion scripts

--- a/gio/meson.build	2018-08-05 21:03:48.163077681 -0500
+++ b/gio/meson.build	2018-08-05 22:27:00.498900375 -0500
@@ -228,7 +228,9 @@
 )
 
 # Generate gdbus-codegen
-subdir('gdbus-2.0/codegen')
+# subdir('gdbus-2.0/codegen')
+
+gdbus_codegen = find_program('gdbus-codegen', required: true)
 
 # Generate xdp-dbus.{c,h}
 xdp_dbus_generated = custom_target('xdp-dbus',
@@ -236,7 +238,6 @@
              'org.freedesktop.portal.OpenURI.xml',
              'org.freedesktop.portal.ProxyResolver.xml'],
     output : ['xdp-dbus.h', 'xdp-dbus.c'],
-    depend_files : gdbus_codegen_built_files,
     command : [python, gdbus_codegen,
                '--interface-prefix', 'org.freedesktop.portal.',
                '--output-directory', '@OUTDIR@',
@@ -256,7 +257,6 @@
 gdbus_daemon_generated = custom_target('gdbus-daemon-generated',
     input : ['dbus-daemon.xml'],
     output : ['gdbus-daemon-generated.h', 'gdbus-daemon-generated.c'],
-    depend_files : gdbus_codegen_built_files,
     command : [python, gdbus_codegen,
                '--interface-prefix', 'org.',
                '--output-directory', '@OUTDIR@',

--- a/docs/reference/gio/meson.build	2018-08-05 21:12:22.275059422 -0500
+++ b/docs/reference/gio/meson.build	2018-08-05 21:13:00.907058050 -0500
@@ -204,7 +204,6 @@
       'gsettings.xml',
       'gresource.xml',
       'gdbus.xml',
-      'gdbus-codegen.xml',
     ],
     expand_content_files : [
       'overview.xml',
@@ -212,7 +211,6 @@
       'migrating-gnome-vfs.xml',
       'migrating-gconf.xml',
       'migrating-gdbus.xml',
-      'gdbus-codegen.xml',
     ],
     html_assets : [
       'gvfs-overview.png',
@@ -232,7 +230,7 @@
 if get_option('man')
   manpages = ['gapplication', 'gio-querymodules', 'glib-compile-schemas',
               'glib-compile-resources', 'gsettings', 'gresource', 'gdbus',
-              'gio', 'gdbus-codegen']
+              'gio']
   foreach page : manpages
     custom_target(page + '-man',
       input: page + '.xml',
