diff --git a/gio/gdbus-2.0/codegen/gdbus-codegen.in b/gio/gdbus-2.0/codegen/gdbus-codegen.in
index 67d367543..55b9e44a0 100755
--- a/gio/gdbus-2.0/codegen/gdbus-codegen.in
+++ b/gio/gdbus-2.0/codegen/gdbus-codegen.in
@@ -50,6 +50,6 @@ if not os.path.isfile(os.path.join(path, 'codegen', 'codegen_main.py')):
     path = os.path.join('@DATADIR@', 'glib-2.0')
 
 sys.path.insert(0, path)
-from codegen import codegen_main
+from gdbus_codegen import codegen_main
 
 sys.exit(codegen_main.codegen_main())
