diff -urN gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in
--- gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-07-12 14:38:49.462666091 -0600
+++ gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-07-12 14:39:30.738665697 -0600
@@ -17,7 +17,7 @@
       <description>If the screen should be dimmed to save power when the computer is idle.</description>
     </key>
     <key name="sleep-inactive-ac-timeout" type="i">
-      <default>1200</default>
+      <default>0</default>
       <summary>Sleep timeout computer when on AC</summary>
       <description>The amount of time in seconds the computer on AC power needs to be inactive before it goes to sleep. A value of 0 means never.</description>
     </key>
@@ -27,7 +27,7 @@
       <description>The type of sleeping that should be performed when the computer is inactive.</description>
     </key>
     <key name="sleep-inactive-battery-timeout" type="i">
-      <default>1200</default>
+      <default>900</default>
       <summary>Sleep timeout computer when on battery</summary>
       <description>The amount of time in seconds the computer on battery power needs to be inactive before it goes to sleep. A value of 0 means never.</description>
     </key>
diff -urN gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in.rej gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in.rej
--- gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in.rej	1969-12-31 17:00:00.000000000 -0700
+++ gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in.rej	2019-07-12 14:40:15.510665270 -0600
@@ -0,0 +1,20 @@
+--- data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-06-19 05:11:41.000000000 -0600
++++ data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-07-12 14:35:10.546668179 -0600
+@@ -12,7 +12,7 @@
+       <description>If the screen should be dimmed to save power when the computer is idle.</description>
+     </key>
+     <key name="sleep-inactive-ac-timeout" type="i">
+-      <default>1200</default>
++      <default>0</default>
+       <summary>Sleep timeout computer when on AC</summary>
+       <description>The amount of time in seconds the computer on AC power needs to be inactive before it goes to sleep. A value of 0 means never.</description>
+     </key>
+@@ -22,7 +22,7 @@
+       <description>The type of sleeping that should be performed when the computer is inactive.</description>
+     </key>
+     <key name="sleep-inactive-battery-timeout" type="i">
+-      <default>1200</default>
++      <default>900</default>
+       <summary>Sleep timeout computer when on battery</summary>
+       <description>The amount of time in seconds the computer on battery power needs to be inactive before it goes to sleep. A value of 0 means never.</description>
+     </key>
