diff -urN gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in
--- gnome-settings-daemon-3.32.1/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-07-12 14:25:53.334673493 -0600
+++ gnome-settings-daemon-3.32.1-fixed/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2019-07-12 14:27:31.810672553 -0600
@@ -6,6 +6,11 @@
       <summary>The brightness of the screen when idle</summary>
       <description>This is the laptop panel screen brightness used when the session is idle.</description>
     </key>
+    <key name="manage-dpms-defaults" type="b">
+      <default>false</default>
+      <summary>Reset X DPMS values</summary>
+      <description>Whether DPMS values will be adjusted by gnome-settings-daemon</description>
+    </key>
     <key name="idle-dim" type="b">
       <default>true</default>
       <summary>Dim the screen after a period of inactivity</summary>
diff -urN gnome-settings-daemon-3.32.1/plugins/power/gpm-common.c gnome-settings-daemon-3.32.1-fixed/plugins/power/gpm-common.c
--- gnome-settings-daemon-3.32.1/plugins/power/gpm-common.c	2019-06-19 05:11:41.000000000 -0600
+++ gnome-settings-daemon-3.32.1-fixed/plugins/power/gpm-common.c	2019-07-12 14:26:25.578673185 -0600
@@ -294,6 +294,14 @@
         return TRUE;
 }
 
+gboolean
+manage_dpms (void)
+{
+        GSettings *settings;
+        settings = g_settings_new ("org.gnome.settings-daemon.plugins.power");
+        return g_settings_get_boolean (settings, "manage-dpms-defaults");
+}
+
 guint
 gsd_power_enable_screensaver_watchdog (void)
 {
@@ -304,7 +312,7 @@
          * way. The defaults are now applied in Fedora 20 from
          * being "0" by default to being "600" by default */
         gdk_error_trap_push ();
-        if (DPMSQueryExtension(GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), &dummy, &dummy))
+        if (manage_dpms () && DPMSQueryExtension(GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), &dummy, &dummy))
                 DPMSSetTimeouts (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), 0, 0, 0);
         gdk_error_trap_pop_ignored ();
         id = g_timeout_add_seconds (XSCREENSAVER_WATCHDOG_TIMEOUT,
diff -urN gnome-settings-daemon-3.32.1/plugins/power/gpm-common.h gnome-settings-daemon-3.32.1-fixed/plugins/power/gpm-common.h
--- gnome-settings-daemon-3.32.1/plugins/power/gpm-common.h	2019-06-19 05:11:41.000000000 -0600
+++ gnome-settings-daemon-3.32.1-fixed/plugins/power/gpm-common.h	2019-07-12 14:26:25.578673185 -0600
@@ -35,6 +35,7 @@
 gboolean         gsd_power_is_hardware_a_tablet         (void);
 guint            gsd_power_enable_screensaver_watchdog  (void);
 void             reset_idletime                         (void);
+gboolean         manage_dpms                            (void);
 
 /* Backlight helpers */
 
diff -urN gnome-settings-daemon-3.32.1/plugins/power/gsd-power-manager.c gnome-settings-daemon-3.32.1-fixed/plugins/power/gsd-power-manager.c
--- gnome-settings-daemon-3.32.1/plugins/power/gsd-power-manager.c	2019-06-19 05:11:41.000000000 -0600
+++ gnome-settings-daemon-3.32.1-fixed/plugins/power/gsd-power-manager.c	2019-07-12 14:26:25.578673185 -0600
@@ -1030,6 +1030,9 @@
         gboolean ret;
         GError *error = NULL;
 
+        if (!(manage_dpms ()))
+                return;
+
         iio_proxy_claim_light (manager, TRUE);
         ret = gnome_rr_screen_set_dpms_mode (manager->priv->rr_screen,
                                              GNOME_RR_DPMS_ON,
@@ -1049,6 +1052,9 @@
         gboolean ret;
         GError *error = NULL;
 
+        if (!(manage_dpms ()))
+               return;
+
         iio_proxy_claim_light (manager, FALSE);
         ret = gnome_rr_screen_set_dpms_mode (manager->priv->rr_screen,
                                              GNOME_RR_DPMS_OFF,

