--- a/src/contacts-window.vala	2019-03-12 15:15:11.000000000 -0500
+++ b/src/contacts-window.vala	2019-06-26 13:46:52.172851432 -0500
@@ -35,9 +35,9 @@
   [GtkChild]
   private TitleBar titlebar;
   [GtkChild]
-  private HeaderBar left_header;
+  private Gtk.HeaderBar left_header;
   [GtkChild]
-  private HeaderBar right_header;
+  private Gtk.HeaderBar right_header;
   [GtkChild]
   private Overlay notification_overlay;
   [GtkChild]
