diff --git a/src/gcal-utils.c b/src/gcal-utils.c
index 36117315..ed5c1a7f 100644
--- a/src/gcal-utils.c
+++ b/src/gcal-utils.c
@@ -211,7 +211,6 @@ gcal_dup_icaltime (const icaltimetype *date)
   new_date->hour = date->hour;
   new_date->minute = date->minute;
   new_date->second = date->second;
-  new_date->is_utc = date->is_utc;
   new_date->is_date = date->is_date;
   new_date->is_daylight = date->is_daylight;
   new_date->zone = date->zone;
