diff -Nuar /var/tmp/portage/sys-power/upower-0.99.8.20171128/work/upower-0.99.3/configure.ac /var/tmp/portage/sys-power/upower-0.99.8.20171128/work/upower-0.99.3-fixed/configure.ac
--- a/configure.ac	2018-11-30 19:40:18.055403653 +0000
+++ b/configure.ac	2018-11-30 19:41:35.408202003 +0000
@@ -2,7 +2,7 @@
 
 m4_define([up_major_version], [0])
 m4_define([up_minor_version], [99])
-m4_define([up_micro_version], [7])
+m4_define([up_micro_version], [8])
 m4_define([up_version],
           [up_major_version.up_minor_version.up_micro_version])

